cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -100,6 +100,31 @@ func (k *proxy) ValidateKubernetesVersion() error {
 	return nil
 }
 
+func (k *proxy) ValidateKubernetesMaxVersion() error {
```

Comment:
Q, have you considered to implement this check into `ValidateKubernetesVersion`, so we get for free the check to be executed in all the commands where we run `EnsureCustomResourceDefinitions` as one of the first operations...

Reply:
See: https://github.com/kubernetes-sigs/cluster-api/pull/4984#discussion_r673707349

If you think it's okay, I can merge the funcs

Reply:
/cc @fabriziopandini  (+dumb question: does it make a difference if I mention someone or not? :)

Reply:
As discussed let's keep it explicit